### PR TITLE
Add dark mode, anchor links, utterances support

### PR DESCRIPTION
**Pull request by opeik**
Hi there,

I've added basic dark mode support as well as some other niceties.

### CODE REVIEW

**Comment by justint** [comment]
Hey, thanks a bunch for the contributions. I should have some time to take a look in the next couple of weeks.

**Comment by justint** [comment]
The dark theming & usage of the CSS custom properties looks great. We should adjust some of the dark theme colors to meet the [AA contrast ratio minimums](https://web.dev/color-and-contrast-accessibility/). I can take a pass at this and push another commit for you to review sometime this week.

Utterances is cool, did not know that existed! I'd love to see it in use with Papaya once you've configured it for your instance, if you plan to use it.

One question I have, is why you chose to replace the existing anchor link?

**Comment by opeik** [comment]
> We should adjust some of the dark theme colors to meet the [AA contrast ratio minimums](https://web.dev/color-and-contrast-accessibility/). I can take a pass at this and push another commit for you to review sometime this week.

Fair point, I'm flat out at the moment so feel free.

> One question I have, is why you chose to replace the existing anchor link?

It was a hot minute ago so I'm struggling to recall 😅 There was some issue, I think in regards to the actual anchor becoming visible on hover?

**Comment by opeik** [comment]
> Are you on a different browser?

Yeah, I use Safari on macOS with Chromium as a backup. That might have been it.

**Comment by justint** [comment]
Ok, I've updated the colors to be AA contrast compliant. 

However, I've just noticed the current code highlighting is a bit of an eyesore in dark mode:
Capture

It's easy to switch to another code theme that works better with dark mode, but it could be tricky to find one theme that looks good in both light and dark mode. I wonder if it's worth supporting custom code themes for either light or dark mode separately, using Zola's [classed syntax highlighting](https://www.getzola.org/documentation/content/syntax-highlighting/#inline-vs-classed-highlighting)?

File: sass/main.scss [hunk 1]
```diff
@@ -588,4 +595,53 @@ footer {
...
+        --text-color: rgb(217, 217, 217);
+        --text-accent-color: rgb(130, 130, 130);
+        --border-color: rgb(66, 66, 66);
+        --link-color: rgb(71, 160, 247);
+        --code-background: rgb(32, 32, 32);
+    }
+
+    .img-dark {
+        filter: invert(100%) hue-rotate(180deg);
+    }
```
Comment (justint) [inline sass/main.scss:646]: What is this new `img-dark` class used for, btw?

Comment (opeik) [reply sass/main.scss:646]: `img-dark` denotes an image can have the colours inverted in dark mode, I used it mostly for diagrams. I thought I added a parameter to the `img` shortcode but looks like I forgot 🥲

**Comment by opeik** [comment]
> I wonder if it's worth supporting custom code themes for either light or dark mode separately, using Zola's [classed syntax highlighting](https://www.getzola.org/documentation/content/syntax-highlighting/#inline-vs-classed-highlighting)?

This sounds like an ideal solution. I think monokai looked *okay* in light mode but having seperate light/dark themes would be really nice!

**Comment by justint** [comment]
Hey @opeik, my apologies for ghosting this thread. I think providing dark/light themes for papaya would indeed be really nice, I'm going to re-open this and look into supporting custom code themes next week.

**Comment by justint** [comment]
Hey @opeik, thanks so much again for these contributions. I've finally found the time to resume work on this and was able to get the syntax highlighting working for light/dark mode.

I'm having some weird issues with getting my changes into this PR, so in case you wish to follow along further I've pushed the changes to a [new branch](https://github.com/justint/papaya/tree/opeik/main) and will be merging into main & releasing soon.

**Pull request closed without merging**